ApiReview/API.Polly.Core/NoDocs/Polly/Outcome.TResult.cs

```diff
@@ -0,0 +1,18 @@
+// Assembly 'Polly.Core'
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.ExceptionServices;
+
+namespace Polly;
+
+public readonly struct Outcome<TResult>
+{
+    public Exception? Exception { get; }
+    public TResult? Result { get; }
+    public bool HasResult { get; }
+    public bool IsVoidResult { get; }
+    public void EnsureSuccess();
```

Comment:
I would expect `Result` getter to throw an exception if there was an exception. 
similar to how Task.Result throws an captured exception.
but is this method even needed?

Reply:
I think we should avoid throwing exceptions from `Outcome` as it is low-level API used in high-perf scenarios and in delegates.

Reply:
This is similar pattern that's used for `HttpResponseMessage` to check for successful status code. Also Cosmos DB SDK has similar implementation - you either work with DTOs directly (higher level) and get the exceptions, or you work with `ResponseMessage` (lower level) but need to check for the result explicitly using `response.EnsureSuccessStatusCode()`.